pkg/gui/commit_message_panel.go

```diff
@@ -51,6 +52,103 @@ func (gui *Gui) handleCommitFocused(g *gocui.Gui, v *gocui.View) error {
 	return gui.renderString(g, "options", message)
 }
 
+type credentials struct {
+	unamePassMessage  string
+	waitForGroup      sync.WaitGroup
+	waitForGroupActie bool
+}
+
+// waitForPassUname wait for a username or password input from the pushPassUname popup
+func (gui *Gui) waitForPassUname(g *gocui.Gui, currentView *gocui.View, passOrUname string) string {
+	gui.credentials.waitForGroupActie = true
+	gui.credentials.waitForGroup.Add(1)
+
+	pushPassUnameView, _ := g.View("pushPassUname")
+	if passOrUname == "username" {
+		pushPassUnameView.Title = gui.Tr.SLocalize("PushUsername")
+		pushPassUnameView.Mask = 0
+	} else {
+		pushPassUnameView.Title = gui.Tr.SLocalize("PushPassword")
+		pushPassUnameView.Mask = '*'
+	}
+	g.Update(func(g *gocui.Gui) error {
+		_, err := g.SetViewOnTop("pushPassUname")
+		if err != nil {
+			return err
+		}
+		err = gui.switchFocus(g, currentView, pushPassUnameView)
+		if err != nil {
+			return err
+		}
+		gui.RenderCommitLength()
+		return nil
+	})
+
+	// wait for username/passwords input
+	gui.credentials.waitForGroup.Wait()
```

Comment:
I think that this might be the culprit. I think the main loop is waiting here and never getting passed. I'm not fully sure why. I'll do some more testing

Reply:
I'll try to set this over to channels those are probably better for doing this